File: low-level-design-java/03_design_patterns/01_creational/03_builder/BuilderPatternDemo.java
public class BuilderPatternDemo {
    public static void main(String[] args) {
        Computer gamingPC = new Computer.Builder("Intel i9", "32GB")
                                .setGraphicsCard(true)
                                .setBluetooth(true)
                                .build();

        Computer officePC = new Computer.Builder("Intel i5", "16GB")
                                .setGraphicsCard(false)
                                .setBluetooth(true)
                                .build();

        System.out.println(gamingPC);
        System.out.println(officePC);
    }
}

File: low-level-design-cpp/03_design_patterns/01_creational/03_builder/BuilderPatternDemo.cpp
int main() {
    Computer gamingPC = Computer::Builder("Intel i9", "32GB")
                            .setGraphicsCard(true)
                            .setBluetooth(true)
                            .build();

    Computer officePC = Computer::Builder("Intel i5", "16GB")
                            .setGraphicsCard(false)
                            .setBluetooth(true)
                            .build();

    gamingPC.printSpecs();
    officePC.printSpecs();

    return 0;
}